sw/airborne/modules/flight_plan_guided/flight_plan_guided.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (C) 2016 - IMAV 2016
+ *
+ * This file is part of Paparazzi.
+ *
+ * Paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * Paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+/**
+ * @file modules/computer_vision/flight_plan_guided.c
+ *
+ * Functions to use directly in a flight plan during an AP_GUIDED flight.
+ *
+ * @author MAVLab IMAV 2016
+ * @author K. N. McGuire 2017
+ *
+ */
+
+#include "modules/flight_plan_guided/flight_plan_guided.h"
+#include "subsystems/ins.h"
+#include "firmwares/rotorcraft/navigation.h"
+#include "firmwares/rotorcraft/guidance/guidance_h.h"
+#include "firmwares/rotorcraft/guidance/guidance_v.h"
+#include "generated/flight_plan.h"
+#include "autopilot.h"
+#include <stdio.h>
+#include <time.h>
+#include "math/pprz_algebra_float.h"
+
+#include "firmwares/rotorcraft/stabilization/stabilization_attitude.h"
+
+
+#include "mcu_periph/uart.h"
+
+// start and stop modules
+#include "generated/modules.h"
+
+float nominal_alt; // nominal flight altitude
+
+//TODO: Find equivelant function in math (couldn't find it yet...)
+
+/* Wraps an angle in radians between -PI and +PI */
+float wrapToPi_guided(float ang)
```

Comment:
This should be removed or rather a float_course_normalize function added to math/pprz_algebra_float.h and maybe a macro analogous to INT32_COURSE_NORMALIZE.